praxis-isncsci/ui
# Send back data to external listener on cell change

Closes #197 

## Problem

Systems communicating with the algorithm via the `external message provider` need to be updated when a value is changed so they can implement features like autosave.

## Solution

- Updated the update cells use case to notify external listeners when cells are changed.
- Close classification panel when values are changed

## Testing

- [x] External message provider is called when values change

<details>
  <summary>Test case</summary>

  1. Navigate to the [external message port provider Storybook story](https://64f8d7c6e093108e99084a70-slmqmvoxza.chromatic.com/?path=/story/app-externalmessageportprovider--primary)
  2. Open the development console
  3. Enter a value (any value should trigger a message including grid values, VAC, DAP, lowest non-key muscles, comments, or star related inputs)
  4. Confirm that in the console you see a _message received from the iFrame_ with the new exam values

  ![image](https://github.com/praxis-isncsci/ui/assets/1294355/05ff6977-10f5-4668-9021-16886d7ff851)

</details>

- [x] Classification dialog is closed when the interface is not `fixed` and a value is changed

<details>
  <summary>Test case</summary>

  1. Navigate to the [test site](https://brave-meadow-05543dc10-198.centralus.4.azurestaticapps.net/)
  2. Complete an exam an press the `calculate` button
  3. Confirm that the classification dialog opens and shows the totals
  4. Enter a new value on the grid
  5. Confirm that the classification dialog is automatically closed

</details>
